specification/keyvault/data-plane/Security.KeyVault.Secrets/models.tsp

```diff
@@ -0,0 +1,389 @@
+   * accepting a Delete operation at the entity level or higher (vault, resource
+   * group, subscription etc.)
+   */
+  Purgeable: "Purgeable",
+
+  /**
+   * Denotes a vault state in which deletion is recoverable, and which also permits
+   * immediate and permanent deletion (i.e. purge). This level guarantees the
+   * recoverability of the deleted entity during the retention interval (90 days),
+   * unless a Purge operation is requested, or the subscription is cancelled. System
+   * wil permanently delete it after 90 days, if not recovered
+   */
+  Recoverable_Purgeable: "Recoverable+Purgeable",
+
+  /**
+   * Denotes a vault state in which deletion is recoverable without the possibility
+   * for immediate and permanent deletion (i.e. purge). This level guarantees the
+   * recoverability of the deleted entity during the retention interval(90 days) and
+   * while the subscription is still available. System wil permanently delete it
+   * after 90 days, if not recovered
+   */
+  Recoverable: "Recoverable",
+
+  /**
+   * Denotes a vault and subscription state in which deletion is recoverable within
+   * retention interval (90 days), immediate and permanent deletion (i.e. purge) is
+   * not permitted, and in which the subscription itself  cannot be permanently
+   * canceled. System wil permanently delete it after 90 days, if not recovered
+   */
+  Recoverable_ProtectedSubscription: "Recoverable+ProtectedSubscription",
+
+  /**
+   * Denotes a vault state in which deletion is recoverable, and which also permits
+   * immediate and permanent deletion (i.e. purge when 7<= SoftDeleteRetentionInDays
+   * < 90). This level guarantees the recoverability of the deleted entity during
+   * the retention interval, unless a Purge operation is requested, or the
+   * subscription is cancelled.
+   */
+  CustomizedRecoverable_Purgeable: "CustomizedRecoverable+Purgeable",
+
+  /**
+   * Denotes a vault state in which deletion is recoverable without the possibility
+   * for immediate and permanent deletion (i.e. purge when 7<=
+   * SoftDeleteRetentionInDays < 90).This level guarantees the recoverability of the
+   * deleted entity during the retention interval and while the subscription is
+   * still available.
+   */
+  CustomizedRecoverable: "CustomizedRecoverable",
+
+  /**
+   * Denotes a vault and subscription state in which deletion is recoverable,
+   * immediate and permanent deletion (i.e. purge) is not permitted, and in which
+   * the subscription itself cannot be permanently canceled when 7<=
+   * SoftDeleteRetentionInDays < 90. This level guarantees the recoverability of the
+   * deleted entity during the retention interval, and also reflects the fact that
+   * the subscription itself cannot be cancelled.
+   */
+  CustomizedRecoverable_ProtectedSubscription: "CustomizedRecoverable+ProtectedSubscription",
+}
+
+/**
+ * The secret set parameters.
+ */
+model SecretSetParameters {
```

Comment:
Isn't this just the subset of properties of a `SecretBundle` that are writeable? If so, why do we need two models and not one with the appropriate `visibility`?

Reply:
This is specified for consistency with the original spec; for at least Go, this is a public model that needs to be consistent.